DSM docs update (#17873)
DataDog/documentation: content/en/data_streams/troubleshooting.md

> ---

Reviewer: Just noting that this page isn't being linked to from any other docs, and it also doesn't appear in the left-nav sidebar (in the `menus.en.yaml` file). Do we want to add it there?

Author: Yes please! Can we follow what other products do? https://a.cl.ly/L1uDEDwp

Similarly on the docs landing page (https://docs.datadoghq.com/), can we add a Data Streams Monitoring tile?

Author: Sounds good on adding the Troubleshooting guide to the left nav.
For the docs landing page, this isn't urgent and I can submit a ticket to the Docs Jira later

Everything in this PR has to be merged today since today is the GA date, the sooner the better though

title: Troubleshooting Data Streams Monitoring
kind: documentation
description: Troubleshoot Data Streams Monitoring
---

{{% site-region region="gov,ap1" %}}
<div class="alert alert-warning">
    Data Streams Monitoring is not available for the {{< region-param key="dd_site_name" >}} site.
</div>
{{% /site-region %}}
  

This page explains common issues with setting up and using Data Streams Monitoring, and how to resolve them. Datadog recommends staying up to date with the latest version of the Datadog tracing libraries you use, as each release contains improvements and fixes.

## Diagnosing common problems

> ### Services aren’t showing up in the DSM Map

Reviewer: ```suggestion
### Services aren't showing up in the DSM Map
```

If you do not see your services in the DSM Map or Overview page after following the [setup instructions][1], ensure these requirements are met: 

> * You’re running the Datadog Agent v7.34.0 or later

Reviewer: ```suggestion
* You're running the Datadog Agent v7.34.0 or later.
```

> * Your service is directly producing or consuming from Kafka or RabbitMQ 

Reviewer: ```suggestion
* Your service is directly producing or consuming from Kafka or RabbitMQ.
```

> * You’re running the following tracing library agent versions:

Reviewer: ```suggestion
* You're running the following tracing library agent versions:
```

   * Java: Agent v1.9.0 or later
   * .NET: Tracer v2.28.0 or later (.NET Core, .NET Framework)
   * Go (manual instrumentation): Data Streams Library v0.2 or later

  

> ### End to end latency metric doesn’t look accurate

Reviewer: ```suggestion
### End-to-end latency metric doesn't look accurate
```

> Latency calculations on a pathway require messages traversing through the pathway to be single threaded. If the messages in your data pipelines are multi-threaded, manual instrumentation is required, which is currently available for [Go applications][2] and [Java applications][3]. If you require manual instrumentation for .NET, contact us.

Reviewer: For "contact us", can we be a little more specific. Is it support they should contact or another group at Datadog?

Author: good catch - could we link here https://docs.datadoghq.com/help/

> In the Pathways tab, if you see the message "latency values may be approximate for these pathways“, view the documentation for [instrumentation guidelines][4].

Reviewer: ```suggestion
In the Pathways tab, if you see the message **latency values may be approximate for these pathways**, view the documentation for [instrumentation guidelines][4].
```

  
### Kafka metrics are missing

> If your application is running in Java, ensure you’re running version v1.9.0 or later of the Java Agent, and that both the producer and the consumer services are instrumented.

Reviewer: ```suggestion
If your application is running in Java, ensure you're running version v1.9.0 or later of the Java Agent, and that both the producer and the consumer services are instrumented.
```

If your applications are running in .NET or Go, the [Kafka Consumer Integration][5] is required for the Kafka metrics to populate.

### RabbitMQ metrics are missing
Ensure the [RabbitMQ integration][6] is set up correctly.
  
### Queue metrics are missing

> The [Kafka integration][7] must be set up for self-hosted, MSK, and Confluent Platform / Cloud environments for metrics to populate in the Queue tab.

Reviewer: ```suggestion
The [Kafka integration][7] must be set up for self-hosted, MSK, and Confluent Platform/Cloud environments for metrics to populate on the Queue tab.
```
